test/integration/src/test/java/io/pravega/test/integration/endtoendtest/EndToEndTruncationTest.java

```diff
@@ -673,7 +674,7 @@ public void testDeleteStreamWhileReading() {
         groupManager.createReaderGroup(readerGroup, ReaderGroupConfig.builder().automaticCheckpointIntervalMillis(500).stream(Stream.of(scope, streamName)).build());
         @Cleanup
         EventStreamReader<String> reader = clientFactory.createReader(String.valueOf(0), readerGroup, new UTF8StringSerializer(), ReaderConfig.builder().build());
-        assertEquals(totalEvents / 2, ReadWriteUtils.readEvents(reader, totalEvents / 2, 0));
+        assertEquals(totalEvents / 2, ReadWriteUtils.readEventsUntil(reader, eventRead -> true, totalEvents / 2, 0));
```

Comment:
sorry, i dont understand the implication here .. when predicate is always true, then we are effectively using the the original method itself.. so what is being fixed here?

Reply:
the previous method stopped reading if a null returned by the reader, which can be due to any reason including a slower response on the test infra. Now the test tries until all the expected events are read or a test timeout is observed.